lib/plugins/setedit/action.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * DokuWiki Plugin setedit (Action Component)
+ *
+ * @license GPL 2 http://www.gnu.org/licenses/gpl-2.0.html
+ * @author  Yvonne Lu <yvonnel@leapinglaptop.com>
+ */
+
+// must be run within Dokuwiki
+if (!defined('DOKU_INC')) die();
+
+if (!defined('DOKU_LF')) define('DOKU_LF', "\n");
+if (!defined('DOKU_TAB')) define('DOKU_TAB', "\t");
+if (!defined('DOKU_PLUGIN')) define('DOKU_PLUGIN',DOKU_INC.'lib/plugins/');
+
+require_once DOKU_PLUGIN.'action.php';
+
+class action_plugin_setedit extends DokuWiki_Action_Plugin {
+
+    public function register(Doku_Event_Handler &$controller) {
+        
+        $controller->register_hook('ACTION_HEADERS_SEND', 'BEFORE', $this, 'handle_set_edit');
+        $controller->register_hook('ACTION_ACT_PREPROCESS', 'BEFORE', $this, 'handle_action_act_preprocess');
+       
+    }
+
+    public function handle_set_edit (Doku_Event &$event, $param) {
+        global $INPUT;
+        global $lang;
+        
+        if(is_array($event->data)){
+            list($act) = array_keys($event->data);
+        } else {
+            $act = $event->data;
+        }
+        //echo "in set_edit act=".$act."<br>";
+        //$plugin_name=trim($INPUT->post->str('plugin'));
+        //if ((strlen($plugin_name)===0) || (strcmp($plugin_name, 'setedit')!==0)){
+        //    return true; //not called from setedit plugin
+        //}
+        
+        if (!isset($_REQUEST['setedit'])) return;   // first time - nothing to do
+        if (!checkSecurityToken()) return;
+        
+        if (isset($_POST['select_edit'])){
+            $select_edit = $_POST['select_edit'];
+            if ($select_edit==='dw'){
+                //$this->editor='dw';
```

Comment:
Again, if these line that are commented out are not going to be used, they should be removed, otherwise a comment is helpful.